docs/guide/en/form-model.md

```diff
@@ -0,0 +1,351 @@
+# Form model
+
+Form model is an abstraction over HTML forms. While you can use [forms](https://github.com/yiisoft/form) directly, 
+it can be more convenient to define form in object-oriented style. With this approach, form fields are defined as class
+properties. Besides describing form data, form model also handles presentation and validation aspects.
+
+To define a form model, create a class extending from `Yiisoft\FormModel\FormModel`.
+
+## Properties
+
+Form model properties are defined as class properties. You can also add getters and setters for each of them. 
+
+```php
+final class LoginForm extends FormModel
+{
+    private ?string $login = null;
+    private ?string $password = null;
+    private bool $rememberMe = false;
+
+    public function getLogin(): ?string
+    {
+        return $this->login;
+    }
+
+    public function getPassword(): ?string
+    {
+        return $this->password;
+    }
+
+    public function getRememberMe(): bool
+    {
+        return $this->rememberMe;
+    }
+
+    public function login(string $value): void
+    {
+        $this->login = $value;
+    }
+
+    public function password(string $value): void
+    {
+        $this->password = $value;
+    }
+
+    public function rememberMe(bool $value): void
+    {
+        $this->rememberMe = $value;
+    }   
+}
+```
+
+Example of working with properties individually:
+
+```php
+use Yiisoft\FormModel\FormModel;
+
+/** @var FormModel $form */
+$form = new FormModel();
+$form->login('john');
+
+$form->hasProperty('login'); // true
+$form->hasProperty('passwordConfirmation'); // false
+
+$form->getLogin(); // "john"
+// or
+$form->getPropertyValue('login'); // "john"
+
+$form->getPropertyValue('password'); // null
+$form->getPropertyValue('passwordConfirmation'); // null
+```
+
+> Static properties are not included in form model's set of properties. On attempt to get such property's value,
+> `Yiisoft\FormModel\Exception\StaticObjectPropertyException` exception will be thrown.
+
+## Meta data
+
+Form model's meta data includes:
+
+- Property labels - the labels associated with each input (visible to the end user). When not set, they will be 
+automatically generated using inflector from the [strings](https://github.com/yiisoft/strings) package.
```

Comment:
Worth linking to exact method.

Reply:
To where exactly? Guide doesn't have dedicated section for method, only for group of methods or `Inflector` in general. Or do you mean to add link to the method's source code?

Reply:
I've meant source code.